### PR TITLE
[WIP] Add gemnasium alert counts to the repositories

### PR DESCRIPTION
**Pull request by jonrober**
Code and test complete, but I don't think I feel too enthusiastic about the UI.  I'm doing it as just another badge at the end of the badge list, that links to the alerts page.  Any thoughts on how it'd be most useful?

### CODE REVIEW

**Comment by tallenaz** [comment]
What do folks think about making the alerts count show up as a tooltip when you hover over the gemnasium badge?

**Comment by jonrober** [comment]
The only problem there is that there are some things we don't mark as Gemnasium currently that are actually there and have the alerts pop up.  (The badge is keying off rails or ruby projects and nothing else.) Of course the fix for that is to improve that check. :)

It sounds good here and I can make that fix/change.

**Comment by jonrober** [comment]
Changes made, which also lets mods_display_app (as an example) to link to its gemnasium page.

File: app/models/gemnasium_client.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'rest-client'
+require 'timeout'
+
+class GemnasiumClient
+  def initialize
+    api_key = Settings.GEMNASIUM_API_TOKEN
+    @api = 'https://X:' + api_key + '@api.gemnasium.com/v1'
```
Comment (tallenaz) [inline app/models/gemnasium_client.rb]: Can we set `@api` up with an attr_reader? That way, when the api is invoked in other methods in the class, we wouldn't need to use the `@`. For an example of the pattern, see https://github.com/sul-dlss/quimby/blob/master/app/models/puppetdb_client.rb#L6-L15

File: app/models/gemnasium_client.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  def all_projects_and_counts
+    advisories = {}
+    slugs = all_projects
+    slugs.shuffle.each do |slug|
+      alerts = fetch_api(@api + '/projects/' + slug + '/alerts')
```
Comment (tallenaz) [inline app/models/gemnasium_client.rb]: Here, and throughout, I'd prefer string interpolation with double quotes to concatenating with single quotes. For example: `"#{api}/projects/#{slug}/alerts"`

File: app/models/gemnasium_client.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  def all_projects
+    slugs = []
+    url = @api + '/projects'
+    projects = fetch_api(url)
+    projects['owned'].each do |project|
+      next unless project['monitored']
+      slugs.push(project['slug'])
+    end
+
```
Comment (tallenaz) [inline app/models/gemnasium_client.rb]: Unnecessary whitespace

File: db/migrate/20170728202534_add_gemnasium_field_to_repositories.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddGemnasiumFieldToRepositories < ActiveRecord::Migration[5.1]
```
Comment (tallenaz) [inline db/migrate/20170728202534_add_gemnasium_field_to_repositories.rb]: Instead of adding a new migration to change Repositories, how about editing the initial Repository migration and schema directly -- in the interest of cleaner migration history, at least until the 1.0 release?

**Comment by jonrober** [comment]
Made the requested changes all.  Thanks for the attr_reader mention, I was wondering why using without the @ was failing.
